packages/grid/x-grid-modules/src/components/checkbox-renderer.tsx

```diff
@@ -4,9 +4,14 @@ import { SelectionChangeParams } from '../models/params/selectionChangeParams';
 import { ColParams } from '../models/params/colParams';
 import { CellParams } from '../models/params/cellParams';
 
-export const HeaderCheckbox: React.FC<ColParams> = React.memo(({ api }) => {
-  const [isChecked, setChecked] = React.useState(false);
-  const [isIndeterminate, setIndeterminate] = React.useState(false);
+export const HeaderCheckbox: React.FC<ColParams> = ({ api }) => {
```

Comment:
We do no longer need the memo?

Reply:
Would it make sense to pull the api from the context here?

Reply:
This is just an implementation of the col prop renderHeader.
So we pass api as a prop so it's easier for the user.
Yes users could use the context as well.